core/arch/interp.c

```diff
@@ -8342,31 +8343,36 @@ fixup_indirect_trace_exit(dcontext_t *dcontext, instrlist_t *trace)
                 /* Add the trace exit label. */
                 instrlist_append(trace, trace_exit_label);
                 /* ldr x0, TLS_REG0_SLOT */
-                instrlist_append(
-                    trace,
-                    instr_create_restore_from_tls(dcontext, scratch, TLS_REG0_SLOT));
+                new_instr =
+                    instr_create_restore_from_tls(dcontext, scratch, TLS_REG0_SLOT);
+                instr_set_translation(new_instr, instr_get_translation(instr));
+                instrlist_append(trace, new_instr);
```

Comment:
You can use INSTR_XL8 to make all of these a single statement again if desired

Reply:
done